### DISCUSSION TITLE
Decoding of nonstandardized-but-common PNG EXIF metadata

### DISCUSSION THREAD

**Discussion by jubilant-enigma** [SixLabors/ImageSharp#1875] [Ideas] [answered]
I think it would be good for ImageSharp to support decoding EXIF metadata stored in PNG images into the ImageMetadata ExifProfile member. This is already supported if the PNG has its exif metadata in the new-as-of-2017 exif chunk. However, storing exif data in PNGs before that point was still desirable, so ImageMagick, ExifTool, exiv2, digiKam, and likely others decided to store the exif metadata blob as hex-encoded text in a PNG chunk named "Raw profile type exif".

It would be helpful to have ImageSharp look for and decode this exif data. I'm not sure if it would be necessary to allow users to save into this chunk as well.

**Comment by JimBobSquarePants** [comment, marked as answer]
As long as the EXIF data follows the standard I have no issue with this.

**Reply by jubilant-enigma** [reply]
Based on some initial testing it seems to be a standard exif blob that can be passed into the constructor of `ExifProfile`.
